test/values.test.js

```diff
@@ -501,12 +538,265 @@ describe('Values', () => {
+          },
+        },
+      }),
+      `Query(Foreach(Paginate(Documents(Collection("orders"))), Lambda("doc", Update(Var("doc"), {data: {createdBy: "Ryan"}}))))`
+    )
+
+    // If
+    assertPrint(
+      new Query({
+        else: 'was false',
+        then: 'was true',
+        if: true,
+      }),
+      `Query(If(true, "was true", "was false"))`
+    )
+
+    // Call
+    assertPrint(
+      new Query({
+        arguments: 2,
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), 2))`
+    )
+
+    assertPrint(
+      new Query({
+        arguments: [1, 2, 3],
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), [1, 2, 3]))`
+    )
+
+    assertPrint(
+      new Query({
+        arguments: {
+          object: {
+            one: 1,
+            two: 2,
+            three: 3,
+          },
+        },
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), {one: 1, two: 2, three: 3}))`
+    )
+
+    // Databases
+    assertPrint(
+      new Query({
+        databases: {
+          database: 'lettuce',
+        },
+      }),
+      'Query(Databases(Database("lettuce")))'
+    )
+
+    assertPrint(
+      new Query({
+        databases: null,
+      }),
+      'Query(Databases())'
+    )
+
+    // Collections
+    assertPrint(
+      new Query({
+        collections: {
+          database: 'lettuce',
+        },
+      }),
+      'Query(Collections(Database("lettuce")))'
+    )
+
+    assertPrint(
+      new Query({
+        collections: null,
+      }),
+      'Query(Collections())'
+    )
+
+    // Documents
+    assertPrint(
+      new Query({
+        documents: {
+          collection: 'lettuce',
+        },
+      }),
+      'Query(Documents(Collection("lettuce")))'
+    )
+
+    assertPrint(
+      new Query({
+        documents: null,
+      }),
+      'Query(Documents(null))'
+    )
+
+    // API v3 Contains* functions
+    assertPrint(
+      new Query({
+        in: { object: { a: { object: { b: 1 } } } },
+        contains_path: ['a', 'b'],
+      }),
+      'Query(ContainsPath(["a", "b"], {a: {b: 1}}))'
+    )
+
+    assertPrint(
+      new Query({
+        in: { object: { a: { object: { b: 1 } } } },
+        contains_field: ['a', 'b'],
+      }),
+      'Query(ContainsField(["a", "b"], {a: {b: 1}}))'
+    )
+
+    assertPrint(
+      new Query({
+        in: { object: { a: { object: { b: 1 } } } },
+        contains_value: ['a', 'b'],
+      }),
+      'Query(ContainsValue(["a", "b"], {a: {b: 1}}))'
+    )
+
+    assertPrint(
+      new Query({
+        let: {
+          three: { add: [{ var: 'two' }, 1] },
+          two: { add: [{ var: 'one' }, 1] },
+          one: { add: [0, 1] },
+        },
+        in: { var: 'three' },
+      }),
+      'Query(Let({three: Add(Var("two"), 1), two: Add(Var("one"), 1), one: Add(0, 1)}, Var("three")))'
```

Comment:
I would add a test with arrays to make sure we are printing it correctly, something like: `Query(Let([{x: 10}, {y: 20}], Add(Var('x'), Var('y'))))`

Reply:
@marrony This query doesn't work in the Webshell. From my reading of the FQL cheatsheet, it seems the second arg to `Let` needs to be a `Var` not an `Add`.

Reply:
the second argument of `Let` is an expression, so it basically can be anything

Reply:
@marrony Got it. I expected it to work, but I am getting this error:
```
Error: [
  {
    "position": [
      "query"
    ],
    "code": "invalid expression",
    "description": "No form/function found, or invalid argument keys: { let, in }."
  }
]
```

Reply:
this worked for me
```javascript
Query(Lambda(x => Let([{y: x}], Add(1, Var("y")))))
```